backend/src/layer/data/database/mealplan_management.rs

```diff
@@ -267,18 +281,96 @@ impl MealplanManagementDataAccess for PersistentMealplanManagementData {
 }
 
 impl PersistentMealplanManagementData {
-    async fn update_food(&self, food_id: Uuid, food_name: &str) -> Result<()> {
+    async fn update_food(
+        &self,
+        food_id: Uuid,
+        food_name: &str,
+        nutrition_data: Option<NutritionData>,
+        environment_info: Option<ParseEnvironmentInfo>,
+    ) -> Result<()> {
         sqlx::query!(
-            "UPDATE food SET name = $2 WHERE food_id = $1",
+            "
+            UPDATE food 
+            SET name = $2 
+            WHERE food_id = $1
+            ",
             food_id,
             food_name
         )
         .execute(&self.pool)
         .await?;
+        self.update_nutrition_data(&food_id, nutrition_data).await?;
+        self.update_environment_data(&food_id, environment_info)
+            .await?;
 
         Ok(())
     }
 
+    async fn update_nutrition_data(
+        &self,
+        food_id: &Uuid,
+        nutrition_data: Option<NutritionData>,
```

Comment:
maybe move the option out of here and call this function conditionally?
This would resemble the semantics more that nothing happens if no new data is provided.
(Is this what we want? if one time the nutrition data is forgotten, we keep the old one; but we can never revert to "no data"---I think this is what we want)